src/Umbraco.Web/Editors/MacrosController.cs

```diff
@@ -97,39 +112,79 @@ public HttpResponseMessage GetById(int id)
+
+            var parameters = macro.Properties.Values
+                                .OrderBy(x => x.SortOrder)
+                                .Select(x => new MacroParameterDisplay()
+                                {
+                                    Editor = x.EditorAlias,
+                                    Key = x.Alias,
+                                    Label = x.Name,
+                                    Id = x.Id
+                                });
+
+            macroDisplay.Parameters = parameters;
+
+            return this.Request.CreateResponse(HttpStatusCode.OK, macroDisplay);
+        }
+
+        [HttpGet]
+        public HttpResponseMessage GetById(Guid id)
+        {
+            var macro = _macroService.GetById(id);
+
+            if (macro == null)
             {
-                parameters.Add(new MacroParameterDisplay
-                {
-                    Editor = param.EditorAlias,
-                    Key = param.Alias,
-                    Label = param.Name,
-                    Id = param.Id
-                });
+                return this.ReturnErrorResponse($"Macro with id {id} does not exist");
             }
 
+            var macroDisplay = Mapper.Map<IMacro, MacroDisplay>(macro);
+
+            var parameters = macro.Properties.Values
+                                .OrderBy(x => x.SortOrder)
+                                .Select(x => new MacroParameterDisplay()
+                                {
+                                    Editor = x.EditorAlias,
+                                    Key = x.Alias,
+                                    Label = x.Name,
+                                    Id = x.Id
+                                });
+
             macroDisplay.Parameters = parameters;
 
             return this.Request.CreateResponse(HttpStatusCode.OK, macroDisplay);
         }
 
+        [HttpGet]
+        public HttpResponseMessage GetById(Udi id)
+        {
+            var guidUdi = id as GuidUdi;
+            if (guidUdi == null)
+                this.ReturnErrorResponse($"Macro with id {id} does not exist");
+
+            var macro = _macroService.GetById(guidUdi.Guid);
+            if (macro == null)
+            {
+                return this.ReturnErrorResponse($"Macro with id {id} does not exist");
+            }
+
+            var macroDisplay = Mapper.Map<IMacro, MacroDisplay>(macro);
+
+            var parameters = macro.Properties.Values
+                                .OrderBy(x => x.SortOrder)
+                                .Select(x => new MacroParameterDisplay()
+                                {
+                                    Editor = x.EditorAlias,
+                                    Key = x.Alias,
+                                    Label = x.Name,
+                                    Id = x.Id
+                                });
+
+            macroDisplay.Parameters = parameters;
```

Comment:
@Shazwazza I guess the `Parameters` could be set in the mapping (similar to how property editors are mapped on datatype), but it is nicer than the existing where it populated `MacroDisplay` properties in the controller.